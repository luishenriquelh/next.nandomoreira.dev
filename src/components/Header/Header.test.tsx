import React from 'react';
import { createRouter } from 'next/router';
import { render } from '@testing-library/react';
import { RouterContext } from 'next/dist/next-server/lib/router-context';
import { Header } from '.';

const router = createRouter('/', {}, '/', {
  initialProps: {},
  pageLoader: jest.fn(),
  App: jest.fn(),
  Component: jest.fn(),
});

describe('<Header />', () => {
  test('Render component', () => {
    const componentTree = render(
      <RouterContext.Provider value={router}>
        <Header />
      </RouterContext.Provider>,
    );
    expect(componentTree).toMatchSnapshot();
  });
});
