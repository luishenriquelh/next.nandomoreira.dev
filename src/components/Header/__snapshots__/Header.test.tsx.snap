// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": .c4 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  white-space: nowrap;
  border: 0;
}

.c4.--focusable:focus,
.c4.--focusable:focus-within {
  position: static;
  width: auto;
  height: auto;
  padding: 0.5rem;
  margin: 0.25rem;
  overflow: visible;
  -webkit-clip: auto;
  clip: auto;
  white-space: normal;
}

.c3 {
  cursor: pointer;
  display: inline-block;
}

.c0 {
  position: absolute;
  left: 0;
  right: 0;
  top: 0;
  z-index: 1000;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 2.5rem;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c6 {
  cursor: pointer;
  display: inline-block;
  font-size: 1rem;
  margin: 0 10px;
  padding: 5px 10px;
  position: relative;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c6::before,
.c6::after {
  content: '';
  display: block;
  position: absolute;
  bottom: 0;
  height: 2px;
  width: 0%;
  opacity: 0.5;
  -webkit-transition: all 0.2s ease-in-out;
  transition: all 0.2s ease-in-out;
}

.c6::before {
  left: 10px;
}

.c6::after {
  right: 10px;
}

.c6.--active::before,
.c6:active::before,
.c6:hover::before,
.c6.--active::after,
.c6:active::after,
.c6:hover::after {
  width: calc(50% - 10px);
}

@media (min-width:768px) {
  .c0 {
    position: fixed;
  }
}

<body>
    <div>
      <header
        class="c0"
      >
        <div
          class="c1"
        >
          <h1
            class="c2"
          >
            <a
              class="c3"
              href="/"
            >
              <span
                class="c4"
              >
                undefinedundefined
              </span>
              <svg
                class=""
                fill="none"
                height="45"
                width="50"
                xmlns="http://www.w3.org/2000/svg"
              >
                <title>
                  undefinedundefined
                </title>
                <path
                  d="M43.247 0v45H32.662L12.597 21.021V45H0V0h10.584L30.65 23.979V0h12.598z"
                />
                <rect
                  class="dot"
                  height="12.857"
                  rx="6.429"
                  width="12.987"
                  x="37.013"
                  y="32.143"
                />
              </svg>
            </a>
          </h1>
          <nav
            class="c5"
          >
            <a
              class="c6"
              href="/sobre"
            >
              Sobre
            </a>
            <a
              class="c6"
              href="/portfolio"
            >
              Portfolio
            </a>
            <a
              class="c6"
              href="/contato"
            >
              Contato
            </a>
            <a
              class="c6"
              href="/cv"
            >
              Resume
            </a>
            <a
              class="c6"
              href="https://blog.nandomoreira.dev"
              rel="noopener noreferrer"
              target="_blank"
            >
              Blog
            </a>
          </nav>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="Headerstyles__Header-sc-1oalovf-0 isOaYz"
    >
      <div
        class="Headerstyles__HeaderInner-sc-1oalovf-1 hVIASu"
      >
        <h1
          class="Headerstyles__Logo-sc-1oalovf-2 kmbipE"
        >
          <a
            class="Logostyles__Logo-ztqmk5-0 jbrqWl"
            href="/"
          >
            <span
              class="SrOnlystyles__Tag-sc-1xgrgn4-0 gepsbO"
            >
              undefinedundefined
            </span>
            <svg
              class="Logostyles__Svg-ztqmk5-1 FzJFq"
              fill="none"
              height="45"
              width="50"
              xmlns="http://www.w3.org/2000/svg"
            >
              <title>
                undefinedundefined
              </title>
              <path
                d="M43.247 0v45H32.662L12.597 21.021V45H0V0h10.584L30.65 23.979V0h12.598z"
              />
              <rect
                class="dot"
                height="12.857"
                rx="6.429"
                width="12.987"
                x="37.013"
                y="32.143"
              />
            </svg>
          </a>
        </h1>
        <nav
          class="Headerstyles__Nav-sc-1oalovf-3 cOClTf"
        >
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 kKSvrN"
            href="/sobre"
          >
            Sobre
          </a>
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 kKSvrN"
            href="/portfolio"
          >
            Portfolio
          </a>
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 kKSvrN"
            href="/contato"
          >
            Contato
          </a>
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 kKSvrN"
            href="/cv"
          >
            Resume
          </a>
          <a
            class="Headerstyles__NavLink-sc-1oalovf-4 kKSvrN"
            href="https://blog.nandomoreira.dev"
            rel="noopener noreferrer"
            target="_blank"
          >
            Blog
          </a>
        </nav>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
